### PR TITLE
Cli fixups

### PR DESCRIPTION
**Pull request by mdlinville**
Two smaller commits address https://github.com/docker/docker.github.io/issues/1451 and remove an unused GIF from `docs/reference/commandline/`.

The big commit standardizes formatting of CLI reference commands:
    
- Command name should be a H1
- Only Description, Examples, and Related Commands should be H2
- Changed 'Related information' heading to 'Related commands' since 99% it is only linking commands
- Added some examples where relevant

These changes will enable us to automate use of the CLI reference files in generated YAML output.

Cute alligators

cc/ @FrenchBen

### CODE REVIEW

File: docs/reference/commandline/attach.md [hunk 1]
```diff
@@ -80,52 +81,72 @@ These `a`, `ctrl-a`, `X`, or `ctrl-\\` values are all examples of valid key
...
-    top - 02:05:52 up  3:05,  0 users,  load average: 0.01, 0.02, 0.05
-    Tasks:   1 total,   1 running,   0 sleeping,   0 stopped,   0 zombie
-    Cpu(s):  0.1%us,  0.2%sy,  0.0%ni, 99.7%id,  0.0%wa,  0.0%hi,  0.0%si,  0.0%st
-    Mem:    373572k total,   355560k used,    18012k free,    27872k buffers
-    Swap:   786428k total,        0k used,   786428k free,   221740k cached
+### Attach to and detach from a running container
 
-    PID USER      PR  NI  VIRT  RES  SHR S %CPU %MEM    TIME+  COMMAND
-     1 root      20   0 17200 1116  912 R    0  0.3   0:00.03 top
+```bash
```
Comment (AkihiroSuda) [inline docs/reference/commandline/attach.md]: I wonder `console` is preferred over `bash` for better rendering, but it can be other PR.

Comment (thaJeztah) [inline docs/reference/commandline/attach.md]: Highlighting will probably improve once we can do https://github.com/docker/docker.github.io/pull/1021, unfortunately that's currently blocked by changes upstream

File: docs/reference/commandline/rm.md [hunk 2]
```diff
@@ -29,40 +29,68 @@ Options:
...
 
-    $ docker rm --link /webapp/redis
-    /webapp/redis
+```bash
+$ docker rm /redis
+
+/redis
+```
+
+### Remove a container and its links
```
Comment (thaJeztah) [inline docs/reference/commandline/rm.md]: Actually, this only removes the "link" (if it's a legacy link on the default bridge network; the same may not apply when using `--link` on a custom network)

File: docs/reference/commandline/rm.md [hunk 3]
```diff
@@ -29,40 +29,68 @@ Options:
...
 
 This will remove the underlying link between `/webapp` and the `/redis`
 containers removing all network communication.
+```bash
+$ docker rm --link /webapp/redis
+
+/webapp/redis
+```
+
+### Force-remove a running container
```
Comment (thaJeztah) [inline docs/reference/commandline/rm.md]: For a follow up; we may want to express the dangers of using `-f`, as it not only forces removal of a container is still running, but also can result in layers being left behind if they fail to be removed

File: docs/reference/commandline/wait.md [hunk 4]
```diff
@@ -23,3 +23,36 @@ Block until one or more containers stop, then print their exit codes
 Options:
       --help   Print usage
 ```
+
+> **Note**: `docker wait` returns `0` for a non-interactive container, even if
+> the container is still running.
```
Comment (thaJeztah) [inline docs/reference/commandline/wait.md]: Can you check this to be sure (per our discussion)

**Review by thaJeztah — CHANGES_REQUESTED** [review]
thanks! left some comments

File: docs/reference/commandline/attach.md (hunk 1 above)
Comment (mstanleyjones) [inline docs/reference/commandline/attach.md]: Either `bash` is a synonym for `console` or the other way around. They are equivalent. We have pretty much standardized on `bash` over time.

File: docs/reference/commandline/dockerd.md [hunk 5]
```diff
@@ -1118,7 +1095,7 @@ If you are running a prometheus server you can add this address to your scrape c
 to have prometheus collect metrics on Docker.  For more information
 on prometheus you can view the website [here](https://prometheus.io/).
 
-```yml
+```none
```
Comment (mstanleyjones) [inline docs/reference/commandline/dockerd.md]: Correct. It screws up the formatting in the whole rest of the file. :(

**Comment by mstanleyjones** [comment]
Addressed feedback so far, except for follow-up issues.

**Review by thaJeztah — APPROVED** [review]
LGTM, but needs a rebase

**Comment by mstanleyjones** [comment]
Rebase complete, @thaJeztah

**Comment by thaJeztah** [comment]
Let's go!

**Pull request merged**